File: leetcode-submissions/1669_Merge_In_Between_Linked_Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeInBetween(ListNode* list1, int a, int b, ListNode* list2) {
     //in this code we have a simple  approach
     ListNode* temp1=list1;
     ListNode* prev=list1;
     ListNode* curr=list1;
     a--;
     while(temp1!=nullptr and a--){
        temp1=temp1->next;
        prev=temp1;
     }
     temp1=list1;
     while(temp1!=NULL and b--){
        temp1=temp1->next;
        curr=temp1;
     }
     ListNode* temp2=list2;
     while(temp2->next!=NULL){
        temp2=temp2->next;
     }
     prev->next=list2;
     temp2->next=curr->next;
     return list1;
    }
};

File: leetcode-submissions/1669_Merge_In_Between_Linked_Lists.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function mergeInBetween(list1: ListNode | null, a: number, b: number, list2: ListNode | null): ListNode | null {
// we need to find the ath node and also the bth node and we need to insert the second list in the gap of the a andb and then return the answer
if(!list2) return list1 ;
let tempNode :ListNode | null =list1;
let athNode :ListNode | null =null;
let bthNode :ListNode | null =null;


let index:number =0;
while(tempNode){
    if(index==a-1){
        athNode=tempNode ;
    }
    if(index==b){
        bthNode=tempNode.next;

    }
    tempNode=tempNode.next;
    index++;
}
let list2End:ListNode | null =list2;
while(list2End.next){
    list2End=list2End.next;
}
if(athNode && bthNode){
    athNode.next=list2;
    if(list2End){
        list2End.next=bthNode;
    }
}
return list1;

};